### PR TITLE
⬆️ Regenerate Protobuf files with new versions of CosmJS and LikeCoin chain

### PR DESCRIPTION
**Pull request by nnkken**
TODO: amino converters

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -22,6 +22,7 @@
   "author": "Wei <wei@liker.land>",
```
Comment (williamchong) [inline package.json:25]: You seems to have used a node version that causes package lock file to upgrade to v2
Please state your node.js version in package.json using `engine` field

File: scripts/1_clone_cosmos_proto.sh [hunk 2]
```diff
@@ -8,23 +8,19 @@ cd "${WD}"
 set -o errexit -o nounset -o pipefail
 command -v shellcheck >/dev/null && shellcheck "$0"
 
-COSMOS_SDK_REF='v0.45.6'
+PROTO_DEPS_DIR="../proto_deps"
+PROTO_DEPS_DIR_FULL_PATH=$(realpath --no-symlink "$PROTO_DEPS_DIR")
+if [ -e "$PROTO_DEPS_DIR_FULL_PATH" ]; then
+  echo "Output path $PROTO_DEPS_DIR_FULL_PATH exists, please remove it first"
+  exit 1
+fi
```
Comment (williamchong) [inline scripts/1_clone_cosmos_proto.sh:16]: Any reason we don't gitignore `proto_deps` if we have this script?

File: scripts/2_generate_typescript.sh [hunk 3]
```diff
@@ -8,60 +8,21 @@ cd "${WD}"
...
 
-ROOT_PROTO_DIR="../cosmos/cosmos-sdk"
-COSMOS_PROTO_DIR="$ROOT_PROTO_DIR/proto"
-THIRD_PARTY_PROTO_DIR="$ROOT_PROTO_DIR/third_party/proto"
+PROTO_DEPS_DIR="../proto_deps"
 PROTO_ROOT_DIR="../proto"
-NFT_PROTO_DIR="../proto/backport"
-ISCN_PROTO_DIR="../proto/likechain"
 OUT_DIR="../src/"
+# OUT_DIR="./output"
```
Comment (williamchong) [inline scripts/2_generate_typescript.sh]: please remove if not needed

File: .gitignore [hunk 4]
```diff
@@ -37,7 +37,4 @@ Thumbs.db
 # Ignore built ts files
 dist/**/*
 
-cosmos
 dist
-src/cosmos/
-src/google/
```
Comment (williamchong) [inline .gitignore:43]: Not sure if we want to commit all cosmos sdk proto into this repo

**Review by williamchong — CHANGES_REQUESTED** [review]

File: scripts/1_clone_cosmos_proto.sh (hunk 2 above)
Comment (nnkken) [reply scripts/1_clone_cosmos_proto.sh:16]: I prefer including dependencies so things are more deterministic.
May remove them if you prefer the other way.

Comment (williamchong) [reply scripts/1_clone_cosmos_proto.sh:16]: Does this script always generate same proto or is it dependent on some env?

Comment (nnkken) [reply scripts/1_clone_cosmos_proto.sh:16]: It use `buf` to download the proto files from Cosmos SDK Buf repo: https://buf.build/cosmos/cosmos-sdk
So I guess it falls into `dependent on some env`
That said, given that the commit is fixed (in the `COSMOS_SDK_PROTO_COMMIT` variable in the script), I guess it is always the same

File: src/likechain/likenft/v1/amino.ts [hunk 5]
```diff
@@ -18,6 +18,21 @@ import {
...
+	CreateListingAuthorization,
+	UpdateListingAuthorization,
+	DeleteListingAuthorization,
+	CreateOfferAuthorization,
+	UpdateOfferAuthorization,
+	DeleteOfferAuthorization,
+	NewClassAuthorization,
+	UpdateClassAuthorization,
+	SendNFTAuthorization,
+	MintNFTAuthorization,
```
Comment (williamchong) [inline src/likechain/likenft/v1/amino.ts]: Why tab here?

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/likechain/likenft/v1/amino.ts (hunk 5 above)
Comment (nnkken) [reply src/likechain/likenft/v1/amino.ts]: Copied from .go file🌚 Let me fix it

**Pull request merged**